api/src/main/java/marquez/db/LineageDao.java

```diff
@@ -79,6 +93,39 @@ SELECT DISTINCT ON (j.uuid) j.*, inputs AS input_uuids, outputs AS output_uuids
   """)
   Set<JobData> getLineage(@BindList Set<UUID> jobIds, int depth);
 
+  /**
+   * 1 level of lineage for all the children jobs of the given parent
+   *
+   * @param parentJobNamespace the namespace of the parent
+   * @param parentJobName the name of the parent
+   * @return edges form job to dataset to job
+   */
+  @SqlQuery(
+      """
+      SELECT
+          jobs.namespace_name AS job_namespace, jobs.simple_name AS job_name,
+          jvim.io_type AS io1,
+          d.namespace_name AS ds_namespace, d."name" AS ds_name,
+          jvim2.io_type AS io2,
+          jv2.namespace_name AS job2_namespace, jv2.job_name  AS job2_name,
+          pj.namespace_name AS job2_parent_namespace, pj.simple_name AS job2_parent_name
+      FROM jobs
+      INNER JOIN job_versions jv ON jv.uuid = jobs.current_version_uuid
+      LEFT JOIN job_versions_io_mapping jvim ON jvim.job_version_uuid = jobs.current_version_uuid
+      LEFT JOIN datasets d ON d.uuid = jvim.dataset_uuid
+      LEFT JOIN job_versions_io_mapping jvim2 ON jvim2.dataset_uuid = d.uuid AND jvim2.job_version_uuid <> jvim.job_version_uuid and jvim2.io_type <> jvim.io_type
+      LEFT JOIN job_versions jv2 ON jv2.uuid = jvim2.job_version_uuid
+      LEFT JOIN jobs j2 ON jv2.job_uuid = j2.uuid
+      LEFT JOIN jobs pj ON j2.parent_job_uuid = pj.uuid
+      WHERE jobs.parent_job_uuid IN (
```

Comment:
How does it differ to current lineage endpoint? Isn't this just `get jobIds of jobs with parent_job X` and `get lineage with depth 1 for those jobIds`. I understand the difference at the REST api, but would love to understand better why do we need separate DAO methods for that. 

Please also be aware of parallel effort to add `job_id` column to `job_versions_io_mapping` (https://github.com/MarquezProject/marquez/pull/2654/files) which limits amount of joins required for this query.

Reply:
This is a good point. I could retrieve the list of jobs for the parent and pass that to the current lineage endpoint with a depth of 0 (or 1?). The new endpoint is trying to fetch less data as well. As the existing lineage endpoint fetched a lot. I can separately add a ?parentNode option to the /lineage endpoint that does what you suggest.

Reply:
As discussed offline, writing single purpose queries may be more efficient although some similarity exists.